tests/ashley/machina_extensions/forum_permission/test_permission_handler.py

```diff
@@ -66,3 +71,58 @@ def test_forum_list(self):
         self.assertNotContains(response, "Forum A2")
         self.assertContains(response, "Forum B1")
         self.assertNotContains(response, "Forum B2")
+
+    def test_get_readable_forums(self):
+        """
+        The get_readable_forums() function should filter the results according
+        to the LTIContext of the user, if available.
+        """
+        user = UserFactory()
+        lti_context_a = LTIContextFactory(lti_consumer=user.lti_consumer)
+        lti_context_b = LTIContextFactory(lti_consumer=user.lti_consumer)
+
+        # Create 2 forums for context A
+        forum_a1 = ForumFactory(name="Forum A1")
+        forum_a1.lti_contexts.add(lti_context_a)
+        forum_a2 = ForumFactory(name="Forum A2")
+        forum_a2.lti_contexts.add(lti_context_a)
+        # Create 2 forums for context B
+        forum_b1 = ForumFactory(name="Forum B1")
+        forum_b1.lti_contexts.add(lti_context_b)
+        forum_b2 = ForumFactory(name="Forum B2")
+        forum_b2.lti_contexts.add(lti_context_b)
+
+        # Grant read-only access for forums A1, A2 and B1 to our user
+        assign_perm("can_see_forum", user, forum_a1, True)
+        assign_perm("can_read_forum", user, forum_a1, True)
+        assign_perm("can_see_forum", user, forum_a2, True)
+        assign_perm("can_read_forum", user, forum_a2, True)
+        assign_perm("can_see_forum", user, forum_b1, True)
+        assign_perm("can_read_forum", user, forum_b1, True)
+
+        # Instantiate the permission Handler
+        permission_handler = PermissionHandler()
+
+        # When the permission handler has no lti context specified,
+        # the get_readable_forums should return all forums the user
+        # has access to
+        forums = Forum.objects.all()
+        readable_forums = permission_handler.get_readable_forums(forums, user)
+
+        self.assertEqual(len(readable_forums), 3)
+        self.assertIn(forum_a1, readable_forums)
+        self.assertIn(forum_a2, readable_forums)
+        self.assertIn(forum_b1, readable_forums)
```

Comment:
I would say `self.assertCountEqual` can do that in one line

Reply:
I forgot this one, thanks!